skbio/embedding/_protein.py

```diff
@@ -81,18 +109,85 @@ def __repr__(self):
         Protein
         """
         seq = Protein(str(self))
-
-        rstr = repr(seq)
-        rstr = rstr.replace("Protein", "ProteinEmbedding")
-        n_indent = 4  # see Sequence.__repr__
-        indent = " " * n_indent
-        rstr = rstr.replace(
-            "has gaps",
-            f"embedding dimension: {self.embedding.shape[1]}\n{indent}has gaps",
+        rstr = _repr_helper(
+            repr(seq), "Protein", "ProteinEmbedding", "embedding",
+            regex_match="has gaps", shape=self.embedding.shape[1]
         )
         return rstr
 
 
 example_protein_embedding = ProteinEmbedding(
     np.random.randn(62, 1024),
```

Comment:
Should use [`np.random.default_rng`](https://numpy.org/doc/stable/reference/random/generator.html) to ensure reproducibility.